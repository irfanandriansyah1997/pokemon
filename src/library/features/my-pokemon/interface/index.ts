export * from './my-pokemon-hooks.interface';
export * from './register-dialog.interface';
