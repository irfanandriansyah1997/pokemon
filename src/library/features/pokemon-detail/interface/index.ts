export * from './pokemon-item.interface';
