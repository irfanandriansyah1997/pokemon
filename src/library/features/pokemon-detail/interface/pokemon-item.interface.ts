/**
 * Pokemon Item Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonItem {
  id: number;
  image: string;
  nickname?: string;
}
