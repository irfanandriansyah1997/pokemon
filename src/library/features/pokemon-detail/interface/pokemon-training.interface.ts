/**
 * Pokemon Training Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTrainingProps {
  title: string;
  value: string;
}
