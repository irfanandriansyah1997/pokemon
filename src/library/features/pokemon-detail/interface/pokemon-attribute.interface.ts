import { IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttribute {
  key: string;
  value: string | string[];
}

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttributeParameter {
  height: number;
  pokeSpecies?: IPokemonSpecies;
  weight: number;
}
