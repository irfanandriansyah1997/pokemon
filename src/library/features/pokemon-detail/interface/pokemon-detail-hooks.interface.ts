import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

import { IPokemonDetailReducer } from './pokemon-detail-reducer.interface';

/**
 * Pokemon Detail Hooks
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooks {
  action: IPokemonDetailHooksAction;
  state: IPokemonDetailReducer;
}

/**
 * Pokemon Detail Hooks Action Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooksAction {
  setPokemon(param: IPokemon | undefined): void;
  setPokemonSpecies(param: IPokemonSpecies): void;
  toggleLoading(param: boolean): void;
}
