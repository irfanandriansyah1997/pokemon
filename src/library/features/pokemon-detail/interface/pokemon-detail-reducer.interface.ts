import { GenActionMap, NullAble } from '@/library/interface/general';
import { IBaseQueryError } from '@/library/interface/gql';
import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Detail Reducer
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailReducer {
  error?: IBaseQueryError;
  isLoadingRest: boolean;
  pokemon?: IPokemon;
  selection: number;
}

/**
 * Pokemon Detail Enum Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export enum IPokemonDetailTypesEnum {
  setLoading = `SET_LOADING`,
  setPokemonState = `SET_POKEMON`,
  setSelection = `SET_SELECTION`,
  setPokemonSpecies = `SET_POKEMON_SPECIES`
}

/**
 * Pokemon Detail Payload Action Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
type IPokemonDetailPayload = {
  [IPokemonDetailTypesEnum.setLoading]: boolean;
  [IPokemonDetailTypesEnum.setPokemonState]: NullAble<IPokemon>;
  [IPokemonDetailTypesEnum.setSelection]: number;
  [IPokemonDetailTypesEnum.setPokemonSpecies]: IPokemonSpecies;
};

export type IPokemonDetailActionType = GenActionMap<IPokemonDetailPayload>[keyof GenActionMap<IPokemonDetailPayload>];
