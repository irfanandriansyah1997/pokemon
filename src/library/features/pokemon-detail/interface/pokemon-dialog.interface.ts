import { IEventComponent } from '@/library/interface/general';
import { IPokemon, IPokemonSprites } from '@/library/interface/pokemon';

export type IEventOnClose = IEventComponent<'on-close'>;

export type IEventOnRelease = IEventComponent<'on-release'>;

export type IPokemonDialogEvent = (
  event: IEventOnClose | IEventOnRelease
) => void;

/**
 * Pokemon Dialog Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDialogProps {
  on: IPokemonDialogEvent;
  pokemon?: IPokemon;
  showDialog: boolean;
}

/**
 * Pokemon Dialog Backdrop Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonDialogBackdropProps {
  color?: string;
  show?: boolean;
}

/**
 * Pokemon Top Section Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTopSectionProps {
  genus: string;
  id: string;
  name: string;
  showImage?: boolean;
  showWrapper?: boolean;
  sprite?: IPokemonSprites;
}
