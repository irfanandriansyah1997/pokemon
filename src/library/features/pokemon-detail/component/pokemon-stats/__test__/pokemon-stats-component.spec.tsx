import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PokemonStats from '@/library/features/pokemon-detail/component/pokemon-stats';
import FullPokemonFixture from '@/library/features/pokemon-detail/reducers/__test__/__fixture__/full-pokemon.fixture.json';
import { IPokemon } from '@/library/interface/pokemon';

describe(`Testing Pokemon Stats Component`, () => {
  it(`Testing Snapshot Test`, () => {
    const ui = shallow(
      <PokemonStats {...((FullPokemonFixture as unknown) as IPokemon)} />
    );

    expect(toJson(ui)).toMatchSnapshot();
  });
});
