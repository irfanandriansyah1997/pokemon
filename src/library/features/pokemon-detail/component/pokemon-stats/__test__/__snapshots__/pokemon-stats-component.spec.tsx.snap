// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon Stats Component Testing Snapshot Test 1`] = `
<Fragment>
  <Styled(p)>
    Basic Stats
  </Styled(p)>
  <Styled(div)
    rowGap={22}
    templates={
      Array [
        "120px",
        "25px",
        "auto",
        "25px",
      ]
    }
  >
    <Styled(p)>
      Hp
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      45
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={45}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
    <Styled(p)>
      Attack
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      49
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={49}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
    <Styled(p)>
      Defense
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      49
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={49}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
    <Styled(p)>
      Spc. Attack
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      65
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={65}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
    <Styled(p)>
      Spc. Defense
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      65
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={65}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
    <Styled(p)>
      Speed
    </Styled(p)>
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      45
    </Styled(p)>
    <ProgressBar
      maxValue={180}
      value={45}
    />
    <Styled(p)
      color="primary"
      fontSize="text"
      fontWeight={500}
    >
      180
    </Styled(p)>
  </Styled(div)>
</Fragment>
`;
